locations/spiders/ultra_liquors.py

```diff
@@ -0,0 +1,30 @@
+import scrapy
+
+from locations.categories import Extras, apply_yes_no
+from locations.dict_parser import DictParser
+from locations.user_agents import BROWSER_DEFAULT
+
+
+class UltraLiquorsSpider(scrapy.Spider):
+    name = "ultra_liquors"
+    item_attributes = {"brand": "Ultra Liquors", "brand_wikidata": "Q116620602"}
+    allowed_domains = ["greenpoint.ultraliquors.co.za"]
+    user_agent = BROWSER_DEFAULT
+    custom_settings = {"ROBOTSTXT_OBEY": False}
+
+    def start_requests(self):
+        yield scrapy.FormRequest(
+            url="https://greenpoint.ultraliquors.co.za/UltraCityStoreSelector/ListData",
+            formdata={"length": "1000"},
+        )
+
+    def parse(self, response, **kwargs):
+        for data in response.json()["Data"]:
+            data["name"] = data.pop("CompanyName")
+            data["street_address"] = data.pop("CompanyAddress")
```

Comment:
The `CompanyAddress` field includes suburb and postcode, hence should be mapped to `addr_full` rather than `street_address`.